lapis2/src/test/kotlin/org/genspectrum/lapis/controller/MultiSegmentedSequenceControllerTest.kt

```diff
@@ -180,49 +145,73 @@ class MultiSegmentedSequenceControllerTest(
                 sequenceFiltersRequest(emptyMap()),
-                SequenceType.UNALIGNED,
-                "otherSegment",
+                SequenceType.ALIGNED,
+                "someSegment",
             )
         } returns returnedValue
 
-        val request = postSample("$UNALIGNED_NUCLEOTIDE_SEQUENCES_ROUTE/otherSegment")
-            .content("""{}""")
-            .contentType(MediaType.APPLICATION_JSON)
-
-        mockMvc.perform(request)
-            .andExpect(status().isOk)
-            .andExpect(content().string(expectedFasta))
-            .andExpect(header().stringValues("Lapis-Data-Version", "1234"))
+        mockMvc.perform(request(UNALIGNED_NUCLEOTIDE_SEQUENCES_ROUTE))
+            .andExpect(status().isNotFound)
     }
 
-    @Test
-    fun `should POST unalignedNucleotideSequences with filter`() {
-        every {
-            siloQueryModelMock.getGenomicSequence(
-                sequenceFiltersRequest(mapOf("country" to "Switzerland")),
-                SequenceType.UNALIGNED,
-                "otherSegment",
-            )
-        } returns returnedValue
-
-        val request = postSample("$UNALIGNED_NUCLEOTIDE_SEQUENCES_ROUTE/otherSegment")
-            .content("""{"country":"Switzerland"}""")
-            .contentType(MediaType.APPLICATION_JSON)
-
-        mockMvc.perform(request)
-            .andExpect(status().isOk)
-            .andExpect(content().string(expectedFasta))
-            .andExpect(header().stringValues("Lapis-Data-Version", "1234"))
+    companion object {
+        @JvmStatic
+        val requestsWithFilter = listOf(
+            Arguments.of(
+                "GET",
+                { route: String ->
+                    getSample(route)
+                        .queryParam("country", "Switzerland")
+                },
+            ),
+            Arguments.of(
+                "POST JSON",
+                { route: String ->
+                    postSample(route)
+                        .content("""{"country": "Switzerland"}""")
+                        .contentType(MediaType.APPLICATION_JSON)
+                },
+            ),
+            Arguments.of(
+                "POST form encoded",
+                { route: String ->
+                    postSample(route)
+                        .param("country", "Switzerland")
+                        .contentType(MediaType.APPLICATION_FORM_URLENCODED_VALUE)
+                },
+            ),
+        )
+
+        @JvmStatic
+        val requestsWithoutFilter = listOf(
+            Arguments.of(
+                "GET",
+                { route: String -> getSample(route) },
+            ),
+            Arguments.of(
+                "POST JSON",
+                { route: String ->
+                    postSample(route)
+                        .content("""{}""")
+                        .contentType(MediaType.APPLICATION_JSON)
+                },
+            ),
+            // Spring doesn't support empty form encoded requests
```

Comment:
Do we need to document this somewhere?

Reply:
Document how? In the form of a test?

Reply:
I thought more maintainer facing. Eg. in swagger you can send an empty request and get the data. But With the forms we should somehow inform the maintainer that empty requests are not working. Maybe in the lapis-2 docu?

Reply:
An empty form is kind of pointless anyway, isn't it? It literally means that there are no values to fill out. 

(Also as a disclaimer: I just tried this from within the Spring tests, not in an e2e scenario).

Reply:
As discussed: sending an empty form without input elements attached to it does not make sense.